wso2as/templates/5.3.0/repository/conf/registry.xml.erb

```diff
@@ -0,0 +1,152 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+
+<!--
+  ~ Copyright 2005-2011 WSO2, Inc. (http://wso2.com)
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~ http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<wso2registry>
+
+    <!--
+    For details on configuring different config & governance registries see;
+    http://wso2.org/library/tutorials/2010/04/sharing-registry-space-across-multiple-product-instances
+    -->
+
+    <currentDBConfig>wso2registry</currentDBConfig>
+    <readOnly>false</readOnly>
+    <enableCache>true</enableCache>
+    <registryRoot>/</registryRoot>
+
+    <!-- This defines the default database and its configuration of the registry -->
+
+    <dbConfig name="wso2registry">
+        <dataSource>jdbc/WSO2CarbonDB</dataSource>
+    </dbConfig>
+
+<%- @master_datasources.each do |datasource_name, datasource| -%>
+  <%- if datasource['mount'] && datasource['mount']['enabled'] == true -%>
+    <dbConfig name="<%= datasource_name %>">
+	<dataSource><%= datasource['jndi_config'] %></dataSource>
+    </dbConfig>
+    <remoteInstance url="https://localhost:9443/registry">
+        <id><%= datasource_name %></id>
+        <dbConfig><%= datasource_name %></dbConfig>
+        <readOnly>false</readOnly>
+        <registryRoot>/</registryRoot>
+        <enableCache>true</enableCache>
+        <cacheId><%= datasource['username'] %>@<%= datasource['url'] %></cacheId>
+    </remoteInstance>
+    <mount path="<%= datasource['mount']['path'] %>" overwrite="virtual">
+        <instanceId><%= datasource_name %></instanceId>
+        <targetPath><%= datasource['mount']['target_path'] %></targetPath>
+    </mount>
+  <%- end -%>
+<%- end -%>
+
+
```

Comment:
We have a separate hiera hash for registry mount. So this registry mounting configuration needs to be re-factored.